{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "adolescent-filling",
   "metadata": {},
   "source": [
    "# Training feature classifier \n",
    "### GTDB ref-seqs and taxonomy were obtained from Qiime2 website and produced by rescript"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f8e14144-bce1-465d-a339-b65a28faeaaf",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.\n",
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.3.1\n",
      "  latest version: 23.5.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.5.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/meco/anaconda3/envs/qiime2-2023.5\n",
      "\n",
      "  added / updated specs:\n",
      "    - ncbi-datasets-pylib\n",
      "    - q2-types-genomics[version='>2023.2']\n",
      "    - xmltodict\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    argh-0.27.2                |     pyhd8ed1ab_0          42 KB  conda-forge\n",
      "    biopython-1.81             |   py38h1de0b5d_0         2.6 MB  conda-forge\n",
      "    gffutils-0.10.1            |     pyh864c0ab_1         1.2 MB  bioconda\n",
      "    jsonlines-3.0.0            |     pyhd8ed1ab_0          12 KB  conda-forge\n",
      "    libprotobuf-3.20.3         |       h3eb15da_0         2.2 MB  conda-forge\n",
      "    ncbi-datasets-pylib-14.25.1|     pyhdfd78af_0         234 KB  bioconda\n",
      "    openssl-3.1.1              |       hd590300_1         2.5 MB  conda-forge\n",
      "    protobuf-3.20.3            |   py38h8dc9893_1         309 KB  conda-forge\n",
      "    pyfaidx-0.7.2.1            |     pyh7cba7a3_1          32 KB  bioconda\n",
      "    pyvcf3-1.0.3               |     pyhdfd78af_0         957 KB  bioconda\n",
      "    q2-types-genomics-2023.5.0.dev0+4.gc4f304a|           py38_0        11.6 MB  https://packages.qiime2.org/qiime2/2023.5/tested\n",
      "    simplejson-3.19.1          |   py38h1de0b5d_0         105 KB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        21.7 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  argh               conda-forge/noarch::argh-0.27.2-pyhd8ed1ab_0 \n",
      "  biopython          conda-forge/linux-64::biopython-1.81-py38h1de0b5d_0 \n",
      "  gffutils           bioconda/noarch::gffutils-0.10.1-pyh864c0ab_1 \n",
      "  jsonlines          conda-forge/noarch::jsonlines-3.0.0-pyhd8ed1ab_0 \n",
      "  libprotobuf        conda-forge/linux-64::libprotobuf-3.20.3-h3eb15da_0 \n",
      "  ncbi-datasets-pyl~ bioconda/noarch::ncbi-datasets-pylib-14.25.1-pyhdfd78af_0 \n",
      "  protobuf           conda-forge/linux-64::protobuf-3.20.3-py38h8dc9893_1 \n",
      "  pyfaidx            bioconda/noarch::pyfaidx-0.7.2.1-pyh7cba7a3_1 \n",
      "  pyvcf3             bioconda/noarch::pyvcf3-1.0.3-pyhdfd78af_0 \n",
      "  q2-types-genomics  qiime2/2023.5/tested/linux-64::q2-types-genomics-2023.5.0.dev0+4.gc4f304a-py38_0 \n",
      "  simplejson         conda-forge/linux-64::simplejson-3.19.1-py38h1de0b5d_0 \n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  openssl                                  3.1.0-hd590300_3 --> 3.1.1-hd590300_1 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "ncbi-datasets-pylib- | 234 KB    |                                       |   0% \n",
      "biopython-1.81       | 2.6 MB    |                                       |   0% \u001b[A\n",
      "\n",
      "pyvcf3-1.0.3         | 957 KB    |                                       |   0% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libprotobuf-3.20.3   | 2.2 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "gffutils-0.10.1      | 1.2 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "pyfaidx-0.7.2.1      | 32 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "protobuf-3.20.3      | 309 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "simplejson-3.19.1    | 105 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "jsonlines-3.0.0      | 12 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "openssl-3.1.1        | 2.5 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "argh-0.27.2          | 42 KB     |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libprotobuf-3.20.3   | 2.2 MB    | ##########7                           |  29% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "ncbi-datasets-pylib- | 234 KB    | ##5                                   |   7% \u001b[A\u001b[A\n",
      "biopython-1.81       | 2.6 MB    | 2                                     |   1% \u001b[A\n",
      "\n",
      "\n",
      "libprotobuf-3.20.3   | 2.2 MB    | #####################7                |  59% \u001b[A\u001b[A\u001b[A\n",
      "biopython-1.81       | 2.6 MB    | #5                                    |   4% \u001b[A\n",
      "\n",
      "ncbi-datasets-pylib- | 234 KB    | ##################################### | 100% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "gffutils-0.10.1      | 1.2 MB    | 4                                     |   1% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "biopython-1.81       | 2.6 MB    | #############8                        |  37% \u001b[A\n",
      "\n",
      "pyvcf3-1.0.3         | 957 KB    | ##################################### | 100% \u001b[A\u001b[A\n",
      "\n",
      "pyvcf3-1.0.3         | 957 KB    | ##################################### | 100% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libprotobuf-3.20.3   | 2.2 MB    | ################################4     |  88% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "gffutils-0.10.1      | 1.2 MB    | ############6                         |  34% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "pyfaidx-0.7.2.1      | 32 KB     | ##################3                   |  50% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "biopython-1.81       | 2.6 MB    | #######################8              |  64% \u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "pyfaidx-0.7.2.1      | 32 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "gffutils-0.10.1      | 1.2 MB    | ########################8             |  67% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "biopython-1.81       | 2.6 MB    | ################################4     |  88% \u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "protobuf-3.20.3      | 309 KB    | #9                                    |   5% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "simplejson-3.19.1    | 105 KB    | #####6                                |  15% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "openssl-3.1.1        | 2.5 MB    | 2                                     |   1% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "protobuf-3.20.3      | 309 KB    | ################################5     |  88% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "jsonlines-3.0.0      | 12 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "openssl-3.1.1        | 2.5 MB    | ###########4                          |  31% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "argh-0.27.2          | 42 KB     | ##############1                       |  38% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "libprotobuf-3.20.3   | 2.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "openssl-3.1.1        | 2.5 MB    | ######################9               |  62% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "gffutils-0.10.1      | 1.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "gffutils-0.10.1      | 1.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "biopython-1.81       | 2.6 MB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | 2                                     |   1% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | 6                                     |   2% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "simplejson-3.19.1    | 105 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "simplejson-3.19.1    | 105 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | #3                                    |   4% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "protobuf-3.20.3      | 309 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "jsonlines-3.0.0      | 12 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | ##7                                   |   7% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | #####5                                |  15% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "argh-0.27.2          | 42 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "argh-0.27.2          | 42 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | #########1                            |  25% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "openssl-3.1.1        | 2.5 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "openssl-3.1.1        | 2.5 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | ###########                           |  30% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | ################5                     |  45% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | ####################7                 |  56% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | #######################5              |  64% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | #########################5            |  69% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | ################################8     |  89% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "q2-types-genomics-20 | 11.6 MB   | ####################################3 |  98% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "                                                                                \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "!conda install -y -c conda-forge -c bioconda -c qiime2 \\\n",
    "    -c https://packages.qiime2.org/qiime2/2023.5/tested/ -c defaults \\\n",
    "    xmltodict 'q2-types-genomics>2023.2' ncbi-datasets-pylib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d851fe9c-a123-499d-ba2b-62b0832a486d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "\n",
      "==> WARNING: A newer version of conda exists. <==\n",
      "  current version: 23.3.1\n",
      "  latest version: 23.5.0\n",
      "\n",
      "Please update conda by running\n",
      "\n",
      "    $ conda update -n base -c defaults conda\n",
      "\n",
      "Or to minimize the number of packages updated during conda update use\n",
      "\n",
      "     conda install conda=23.5.0\n",
      "\n",
      "\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/meco/anaconda3/envs/qiime2-2023.5\n",
      "\n",
      "  added / updated specs:\n",
      "    - git\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    git-2.41.0                 | pl5321h86e50cf_0         9.4 MB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         9.4 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  git                conda-forge/linux-64::git-2.41.0-pl5321h86e50cf_0 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "                                                                                \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "!conda install -y -c conda-forge -c bioconda git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4df3afe8-8090-41a3-8060-9912a9c60ae6",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting git+https://github.com/bokulich-lab/RESCRIPt.git\n",
      "  Cloning https://github.com/bokulich-lab/RESCRIPt.git to /tmp/pip-req-build-cxiapq30\n",
      "  Running command git clone --filter=blob:none --quiet https://github.com/bokulich-lab/RESCRIPt.git /tmp/pip-req-build-cxiapq30\n",
      "  Resolved https://github.com/bokulich-lab/RESCRIPt.git to commit c7c6b4e8b0d7e2f1654806e733de5326757b442f\n",
      "  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hBuilding wheels for collected packages: rescript\n",
      "  Building wheel for rescript (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for rescript: filename=rescript-2023.5.0-py3-none-any.whl size=6318847 sha256=29d4356712a6edb248d0dda0052875f6569f218bfa6015a3d7d7bf169109865b\n",
      "  Stored in directory: /tmp/pip-ephem-wheel-cache-j8ewrai7/wheels/a7/2f/ca/a4cfe2ac81c54ea686727a464c1029266233e0df67566e3523\n",
      "Successfully built rescript\n",
      "Installing collected packages: rescript\n",
      "Successfully installed rescript-2023.5.0\n"
     ]
    }
   ],
   "source": [
    "!pip install git+https://github.com/bokulich-lab/RESCRIPt.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "10228a4f-329a-4758-b0c6-fd801cffd14e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mQIIME is caching your current deployment for improved performance. This may take a few moments and should only happen once per deployment.\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime dev refresh-cache"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "052669db-5886-4f62-87d4-4bfb7e3ba1b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "DOMS = {\n",
    "    'Archaea': ['CAGYGCASCAGKCGMGAA', 'GGACTACVSGGGTATCTAAT'],\n",
    "    'Bacteria': ['CAAGRGTTHGATYMTGGCTCAG', 'TGCTGCCTCCCGTAGGAGT']}\n",
    "\n",
    "outdir = 'Classifier_GTDB'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "about-structure",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Taxonomy] to: Classifier_GTDB/Archaea-gtdb-214.1-tax.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Archaea-gtdb-214.1-seqs.qza\u001b[0m\n",
      "\u001b[0m\u001b[32mSaved FeatureData[Taxonomy] to: Classifier_GTDB/Bacteria-gtdb-214.1-tax.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Bacteria-gtdb-214.1-seqs.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!mkdir -p $outdir\n",
    "\n",
    "for dom in DOMS:\n",
    "    \n",
    "    !qiime rescript get-gtdb-data \\\n",
    "        --p-version '214' \\\n",
    "        --p-domain $dom \\\n",
    "        --o-gtdb-taxonomy $outdir/$dom-gtdb-214-tax.qza \\\n",
    "        --o-gtdb-sequences $outdir/$dom-gtdb-214-seqs.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "flying-annotation",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Archaea-gtdb-214.1-seqs.qza\u001b[0m\n",
      "\u001b[0m\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Bacteria-gtdb-214.1-seqs.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "for dom in DOMS:\n",
    "    \n",
    "    !qiime rescript cull-seqs \\\n",
    "        --i-sequences $outdir/$dom-gtdb-214-seqs.qza \\\n",
    "        --o-clean-sequences $outdir/$dom-gtdb-214-seqs.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "informative-blues",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Archaea-gtdb-214-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Taxonomy] to: Classifier_GTDB/Archaea-gtdb-214-tax.qza\u001b[0m\n",
      "\u001b[0m\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Bacteria-gtdb-214-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Taxonomy] to: Classifier_GTDB/Bacteria-gtdb-214-tax.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "for dom in DOMS:\n",
    "\n",
    "    !qiime rescript dereplicate \\\n",
    "        --i-sequences $outdir/$dom-gtdb-214-seqs.qza  \\\n",
    "        --i-taxa $outdir/$dom-gtdb-214-tax.qza \\\n",
    "        --p-mode 'uniq' \\\n",
    "        --o-dereplicated-sequences $outdir/$dom-gtdb-214-seqs.qza \\\n",
    "        --o-dereplicated-taxa $outdir/$dom-gtdb-214-tax.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "entitled-alfred",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Archaea-gtdb-214-seqs.qza\u001b[0m\n",
      "\u001b[0m\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Bacteria-gtdb-214-seqs.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "for dom in DOMS:\n",
    "    f = DOMS[dom][0]\n",
    "    r = DOMS[dom][1]\n",
    "    \n",
    "    !qiime feature-classifier extract-reads \\\n",
    "        --i-sequences $outdir/$dom-gtdb-214-seqs.qza \\\n",
    "        --p-f-primer $f \\\n",
    "        --p-r-primer $r \\\n",
    "        --p-n-jobs 4 \\\n",
    "        --p-read-orientation 'both' \\\n",
    "        --o-reads $outdir/$dom-gtdb-214-seqs.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "forbidden-french",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Archaea-gtdb-214-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Taxonomy] to: Classifier_GTDB/Archaea-gtdb-214-tax.qza\u001b[0m\n",
      "\u001b[0m\u001b[32mSaved FeatureData[Sequence] to: Classifier_GTDB/Bacteria-gtdb-214-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Taxonomy] to: Classifier_GTDB/Bacteria-gtdb-214-tax.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "for dom in DOMS:\n",
    "\n",
    "    !qiime rescript dereplicate \\\n",
    "        --i-sequences $outdir/$dom-gtdb-214-seqs.qza  \\\n",
    "        --i-taxa $outdir/$dom-gtdb-214-tax.qza \\\n",
    "        --p-mode 'uniq' \\\n",
    "        --o-dereplicated-sequences $outdir/$dom-gtdb-214-seqs.qza \\\n",
    "        --o-dereplicated-taxa $outdir/$dom-gtdb-214-tax.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "offensive-grocery",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved TaxonomicClassifier to: Classifier_GTDB/Archaea-gtdb-214-classifier.qza\u001b[0m\n",
      "\u001b[0m\u001b[32mSaved TaxonomicClassifier to: Classifier_GTDB/Bacteria-gtdb-214-classifier.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "for dom in DOMS:\n",
    "\n",
    "    !qiime feature-classifier fit-classifier-naive-bayes \\\n",
    "        --i-reference-reads $outdir/$dom-gtdb-214-seqs.qza \\\n",
    "        --i-reference-taxonomy $outdir/$dom-gtdb-214-tax.qza \\\n",
    "        --o-classifier $outdir/$dom-gtdb-214-classifier.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "987e48a5-543a-465d-87d5-0e542ce9f335",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:qiime2-2023.5]",
   "language": "python",
   "name": "conda-env-qiime2-2023.5-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
